drivers/pn532/pn532.c

```diff
@@ -222,10 +227,15 @@ static int _read(const pn532_t *dev, uint8_t *buff, unsigned len)
         buff[0] = 0x80;
         reverse(buff, len);
         ret = (int)len + 1;
+        break;
 #endif
+    default:
+        DEBUG("pn532: invalid mode (%i)!\n", dev->mode);
+    }
+    if (ret > 0) {
```

Comment:
This different from the previous behavior. I would just skip it if not using `ENABLE_DEBUG` (I think it should be optimized by the compiler):
```c
if (ENABLE_DEBUG && (ret > 0)) {
...
}
```

Reply:
the compile will optimise this away even without `ENABLE_DEBUG` as extra condition, thus its not needed here.

Reply:
and yes the behaviour is different, but safer because before ifs there was an error but debugging enabled it would print undefined buffer content.